3IPs/3ip-2.md

```diff
@@ -9,92 +9,68 @@ created: 2018-11-20
 
 ## Access control
 
-Right now orbitdb doen't support adding and removing users to a given DB. However if we are fine with the constraint that users can only be added to the thread and not removed we can do the following. For a new thread we create a random seed `R`. From `R` we derive a signing key `k` and a symmetric encryption key `s`. The signing key `k` is given write permissions in the orbitdb feedstore. Each user has a public encryption key `E_<user>` associated with their 3box. If *Alice*  creates the thread, she encrypts `R` to her key `E_alice` and adds this encrypted blob as an entry in the thread. Now when *Alice* want's to grant access to *Bob* she simply encrypts `R` to `E_bob` and adds this encrypted blob as an entry in the thread. *Bob* can now see this encrypted entry and decrypt it to recover `k` and `s` which can be used to post in the thread.
+This first version of threads will allow anyone to post, without any ability to remove posts. This is achieved by using `write: ['*']` when creating the store in orbitdb.
 
-OrbitDB is [working on adding](https://github.com/orbitdb/orbit-db/pull/495) support for access controllers. When they are released we can use them for write access to threads. There is not really any benefit in changing to this until they also add the ability to remove users.
+OrbitDB is [working on adding](https://github.com/orbitdb/orbit-db/pull/495) support for access controllers. When they are released we can use them for write access to threads, as well as moderation, and potentially other features as well.
 
-## Encryption
-
-We simply use `s` to symmetrically encrypt the entries in the thread if it's a private thread.
-
-## Thread subscriptions
-
-Users can subscribe to threads. This means that they will remember the thread next time they go online, and they can ask the network for updates. When a user subscribes to a thread it simply adds the thread-id and name to the rootStore of the users 3Box. When the user unsubscribes the thread entry is removed from the rootStore.
-
-## Threads in spaces?
-
-Is this something we want to enable? Maybe as a second iteration, it adds a lot of complexity.
+## Thread creation
+Threads are created inside of [Spaces](./3ip-1.md). A thread has a `name` which could be any string. For example the name could be the address of a contract if you want to create a thread for commenting on that contract. This also means that two different spaces could have their own threads with the same name. When a thread is created it gets the name `3box.thread.<space-name>.<thread-name>` for the orbitdb logstore.
```

Comment:
What does it mean for a thread to be inside of a space? If the subject of a thread is the address of a contract, for example, would this thread live in the 3Box of the app address that enabled the UI to take place? Or does the root thread live in the user's app space?

Reply:
It means that the access control for writing to the thread is based on the space. Say space `A` and space `B` are used by two different apps, but they both comment on contract `X`. This means that there will be two separate threads, and also that only dapps which the user has allowed to open space `A` can open and write to the threads in that space.

I'm not exactly sure what you mean by "root thread"? All threads are essentially the same.

Reply:
How does this hold `only dapps which the user has allowed to open space A can open and write to the threads in that space`? is there some values from opening the space that derive a different db manifest

Reply:
@zachferland only that the name of the odb store is `3box.thread.<space-name>.<thread-name>`

Reply:
@zachferland only that the name of the odb store is `3box.thread.<space-name>.<thread-name>`

Reply:
so basically if entire name is known, it can be written to?

Reply:
exactly yea :)